build/gradle.go

```
@@ -0,0 +1,250 @@
+	log.SetLogger(containingBuild.logger)
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	extractorLocalPath := filepath.Join(home, dependenciesDirName, "gradle")
+	return &GradleModule{
+		srcPath:         srcPath,
+		containingBuild: containingBuild,
+		gradleExtractorDetails: &gradleExtractorDetails{
+			localPath: extractorLocalPath,
+			tasks:     []string{"aP"},
+			propsDir:  filepath.Join(containingBuild.tempDirPath, PropertiesTempfolderName),
+		},
+	}, err
+}
+
+func (gm *GradleModule) SetExtractorDetails(localExtractorPath, extractorPropsDir string, tasks []string, useWrapper, usePlugin bool, downloadExtractorFunc func(downloadTo, downloadFrom string) error, props map[string]string) *GradleModule {
+	gm.gradleExtractorDetails.tasks = tasks
+	gm.gradleExtractorDetails.propsDir = extractorPropsDir
+	gm.gradleExtractorDetails.useWrapper = useWrapper
+	gm.gradleExtractorDetails.usePlugin = usePlugin
+	gm.gradleExtractorDetails.localPath = localExtractorPath
+	gm.gradleExtractorDetails.downloadExtractorFunc = downloadExtractorFunc
+	gm.gradleExtractorDetails.props = props
+	return gm
+}
+
+// Generates Gradle build-info.
+func (gm *GradleModule) CalcDependencies() error {
+	log.Info("Running gradle...")
+	if gm.srcPath == "" {
+		var err error
+		if gm.srcPath, err = os.Getwd(); err != nil {
+			return err
+		}
+	}
+
+	err := downloadGradleDependencies(gm.gradleExtractorDetails.localPath, gm.gradleExtractorDetails.downloadExtractorFunc)
+	if err != nil {
+		return err
+	}
+	if !gm.gradleExtractorDetails.usePlugin {
+		gradlePluginFilename := fmt.Sprintf(GradleExtractorFileName, GradleExtractorDependencyVersion)
+
+		gm.gradleExtractorDetails.initScript, err = getInitScript(gm.gradleExtractorDetails.localPath, gradlePluginFilename)
+		if err != nil {
+			return err
+		}
+	}
+	gradleRunConfig, err := gm.createGradleRunConfig()
+	if err != nil {
+		return err
+	}
+	defer os.Remove(gradleRunConfig.extractorPropsFile)
```

Let's not ignore the error returned from ```os.Remove```.